packages/ember-routing/lib/system/router.js

```diff
@@ -705,6 +706,24 @@ var EmberRouter = EmberObject.extend(Evented, {
       run.cancel(this._slowTransitionTimer);
     }
     this._slowTransitionTimer = null;
+  },
+
+  _handledErrors: computed(function() {
```

Comment:
If we go down this path, it seems simpler to just brand the given error with a hasBeenHandled symbol. No need to keep these data-structures arround.